### PR TITLE
Adds AGU journals template

### PR DESCRIPTION
**Pull request by eliocamp**
This adds a template for the AGU journals. Tested the skeleton on my machine and it worked. I hope everything else is in order. 
Haven't signed up the Contributor Agreement yet because I don't have access to a printer right now, but will do next week :).

### CODE REVIEW

**Comment by eliocamp** [comment]
I've signed and sent the agreement. :)

**Comment by matthewross07** [comment]
Many thanks @eliocamp , I just pulled this branch for an upcoming submission to WRR. Hope to see this request merged soon.

**Comment by eliocamp** [comment]
They come with the template and is used by the journal to track changes during the submission process, I think. 
The template works perfectly without them.

**Comment by yihui** [comment]
I see. I'd prefer not to have them in the **rticles** package. You could add instructions in `skeleton.Rmd` for those who want to use this feature.

**Comment by eliocamp** [comment]
Ok! I'll be uncomputered until march, I'll do it then!

**Comment by yihui** [comment]
No problem. Thank you!

**Comment by eliocamp** [comment]
I've removed the trackchanges files and added instructions for using it.

File: inst/rmarkdown/templates/agu_article/skeleton/skeleton.Rmd [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+# JGR: Atmospheres, JGR: Biogeosciences, JGR: Earth Surface, 
+# JGR: Oceans, JGR: Planets, JGR: Solid Earth, JGR: Space Physics, 
+# Global Biogeochemical Cycles, Geophysical Research Letters, 
+# Paleoceanography and Paleoclimatology, Radio Science, Reviews of Geophysics, 
+# Tectonics, Space Weather, Water Resources Research, Geochemistry, Geophysics, 
+# Geosystems, Journal of Advances in Modeling Earth Systems (JAMES),
+# Earth's Future, Earth and Space Science, Geohealth
+journal: "JGR: Atmospheres"
+# Use draft to submit a paper
+options: "draft,linenumbers"
```
Comment (yihui) [inline inst/rmarkdown/templates/agu_article/skeleton/skeleton.Rmd]: I'd use `classoption`, following the convention of Pandoc: https://pandoc.org/MANUAL.html#variables-for-latex

File: R/agu_article.R [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#' library(rmarkdown)
+#' draft("MyArticle.Rmd", template = "agu_article", package = "rticles")
+#' }
+#'
+#' @export
+agu_article <- function(..., keep_tex = TRUE, citation_package = 'natbib',
+                        highlight = NULL, md_extensions = c("-autolink_bare_uris",
+                                                            "-auto_identifiers")) {
+  template <- find_resource("agu_article", "template.tex")
+  fmt <- inherit_pdf_document(template = template,
```
Comment (yihui) [inline R/agu_article.R]: `fmt` is not used here.

```suggestion
  inherit_pdf_document(template = template,
```

File: inst/rmarkdown/templates/agu_article/resources/template.tex [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+% Step 1: Set the \documentclass
+%
+% There are two options for article format:
+%
+% PLEASE USE THE DRAFT OPTION TO SUBMIT YOUR PAPERS.
+% The draft option produces double spaced output.
+%
+
+%% To submit your paper:
+\documentclass[$options$]{agujournal2018}
```
Comment (yihui) [inline inst/rmarkdown/templates/agu_article/resources/template.tex]:
```suggestion
\documentclass[$for(classoption)$$classoption$$sep$,$endfor$]{agujournal2018}
```

**Pull request merged**